singular-kit/src/main/java/com/mparticle/kits/SingularKit.java

```diff
@@ -0,0 +1,172 @@
+package com.mparticle.kits;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+
+import com.mparticle.DeepLinkListener;
+import com.mparticle.DeepLinkResult;
+import com.mparticle.MPEvent;
+import com.mparticle.MParticle;
+import com.mparticle.internal.MPUtility;
+import com.singular.sdk.DeferredDeepLinkHandler;
+import com.singular.sdk.Singular;
+import com.singular.sdk.SingularConfig;
+
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONObject;
+
+
+public class SingularKit extends KitIntegration implements KitIntegration.ActivityListener, KitIntegration.EventListener, KitIntegration.PushListener, DeferredDeepLinkHandler {
+
+    private static final String API_KEY = "apiKey";
+    private static final String API_SECRET = "secret";
+    private static final String DDL_TIME_OUT = "ddlTimeout";
+    String singularKey;
+    String singularSecret;
+    SingularConfig config;
+    long DDL_HANDLER_TIMEOUT_SEC = 60L;
+
+    Context context;
+
+    @Override
+    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) {
+        singularKey = settings.get(API_KEY);
+        singularSecret = settings.get(API_SECRET);
+        String ddlTimeout = settings.get(DDL_TIME_OUT);
+        if (!TextUtils.isEmpty(ddlTimeout)) {
+            try {
+                DDL_HANDLER_TIMEOUT_SEC = Long.parseLong(ddlTimeout);
+            } catch (Exception unableToGetDDLTimeout) {
+            }
+        }
+        this.context = context;
+        config = new SingularConfig(singularKey, singularSecret);
+        config.withDDLTimeoutInSec(DDL_HANDLER_TIMEOUT_SEC);
+        config.withDDLHandler(this);
+        Singular.init(context, config);
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> setOptOut(boolean b) {
+        return null;
```

Comment:
Is there a way to disable the singular SDK/opt out? If so, this method should be implemented and should return the proper ReportingMessage.